src/scene/shader-lib/chunks/lit/frag/shadowPCSS.js

```diff
@@ -0,0 +1,185 @@
+export default /* glsl */`
+
+
+#define PCSS_SAMPLE_COUNT 16
+
+vec2 VogelDisk(float sampleIndex, float count, float phi)
+{
+    const float GoldenAngle = 2.4;
+    float r = sqrt(sampleIndex + 0.5) / sqrt(count);
+
+    float theta = sampleIndex * GoldenAngle + phi;
+
+    float sine, cosine;
+    sine = sin(theta);
+    cosine = cos(theta);
+
+    return vec2(r * cosine, r * sine);
+}
+
+vec3 VogelSphere(float sampleIndex, float count, float phi) {
+    float weight = sampleIndex / count;
+    float radius = sqrt(1.0 - weight * weight);
+    const float GoldenAngle = 2.4;
+
+    float theta = GoldenAngle * sampleIndex + phi;
+
+    return vec3(cos(theta) * radius, weight, sin(theta) * radius);
+}
+
+float GradientNoise(vec2 screenPos) {
+    vec3 magic = vec3(0.06711056, 0.00583715, 52.9829189);
+    return fract(magic.z * fract(dot(screenPos, magic.xy)));
+}
+
+#ifndef UNPACKFLOAT
+#define UNPACKFLOAT
+float unpackFloat(vec4 rgbaDepth) {
+    const vec4 bitShift = vec4(1.0 / (256.0 * 256.0 * 256.0), 1.0 / (256.0 * 256.0), 1.0 / 256.0, 1.0);
+    return dot(rgbaDepth, bitShift);
+}
+#endif
+
+float viewSpaceDepth(float depth, mat4 invProjection) {
+    float z = depth * 2.0 - 1.0;
+    vec4 clipSpace = vec4(0.0, 0.0, z, 1.0);
+    vec4 viewSpace = invProjection * clipSpace;
+    return viewSpace.z;
+}
+
+
+float PCSSBlockerDistance(TEXTURE_ACCEPT(shadowMap), vec2 sampleCoords[PCSS_SAMPLE_COUNT], vec2 shadowCoords, float searchSize, float z) {
+
+    float blockers = 0.0;
+    float averageBlocker = 0.0;
+    for (int i = 0; i < PCSS_SAMPLE_COUNT; i++) {
+        vec2 offset = sampleCoords[i] * searchSize;
+        vec2 sampleUV = shadowCoords + offset;
+
+    #ifdef GL2
+        float blocker = texture2D(shadowMap, sampleUV).r;
```

Comment:
see if you can use textureShadow instead of texture2D

Reply:
I need the sampler to be a sampler2D and not samplerShadow, since I need the actual depth and not the binary result from a comparison sample.

Reply:
Done.